### DISCUSSION TITLE
Timeout Argument for Banner type

### DISCUSSION THREAD

**Discussion by SIXfoot5ins** [IBM/mac-ibm-notifications#16] [Ideas]
Hi,
Thanks for this app
Would be great if I could use the timeout argument to control how long a banner notification is on screen for.

I'm using this app to record when users defer my own macOS software updates prompts, I believe the default duration is about 5 seconds, to have it nagging on the screen for 5 minutes so the user has to click it for it to be removed will record the defer action.

Also, if no action is taken on a banner, the notification doesn't seem to exit with a return value, zero or a new number would be useful.

Thanks again for all your help

**Comment by SMartorelli** [comment]
Hi, 
thanks for your feedbacks!

Notification Center’s “Notification” can have two different styles on macOS :
- Banner: lasts 5 sec and then goes away, always visible in the Notification Center history.
- Alert: stuck on top right of the screen as long as the user doesn’t interact with it (like Calendar one for example).

<img width="397" alt="Screenshot 2021-04-08 at 10 33 44" src="https://user-images.githubusercontent.com/23209655/113996009-0d70b900-9857-11eb-9b88-8b8049438c65.png">

This style is set by default to “Banner” for every new app and you can change it in the Notifications panel of System Preferences. There is no way to change it from the app it self but you should change it manually or deploying a configuration profile that does so ([Here](https://developer.apple.com/documentation/devicemanagement/notifications/notificationsettingsitem) and [here](https://www.jamf.com/jamf-nation/articles/648/deploying-custom-configuration-profiles-using-jamf-pro) some docs about).

We’re working on a new feature that will allow the admin to choice which style to use but this require a good amount of effort and we plan to release it by the end of Q3.

Since we currently don’t have the chance to change this behaviour from Mac@IBM Notifications also the -timeout (arguments to set timeout for user interaction) for the “banner” (Mac@IBM Notifications UI style) has been disabled and we suggest to not use “banner” style type for communications that requires mandatory user interaction and that blocks a workflow as long as the user doesn’t act on the banner.

PS: Notification Center Banner’s duration is an unchangeable system default (5 sec).

Hope this can help you!
Best,
SM

**Reply by SIXfoot5ins** [reply]
Thanks for your prompt reply Simone, this makes sense

Can you confirm the issue where if there is not interaction with the banner that the app stays running instead of exiting with a return code like '*** No interaction' for example

**Reply by SMartorelli** [reply]
Yeah it is not an issue in this case since you're maybe still waiting for an user response. If you kill the process from the script the notification will remain in the notification center history panel of the end user and thats why we suggest to use "banner" UI style only for communication that doesn't require mandatory user interactions.

**Reply by SIXfoot5ins** [reply]
Ok, I should be able to work with that in killing the process after 5 seconds

Thanks again for your help and direction

**Reply by SMartorelli** [reply]
Anytime!
